tldw_Server_API/tests/Chat/integration/conftest.py

```
@@ -0,0 +1,82 @@
+        os.environ["OPENAI_API_BASE"] = f"{mock_openai_server}/v1"
+
+        # Patch schema-level API_KEYS so provider is considered configured
+        try:
+            import importlib
+            import tldw_Server_API.app.api.v1.schemas.chat_request_schemas as chat_schemas
+            importlib.reload(chat_schemas)
+            chat_schemas.API_KEYS["openai"] = os.environ["OPENAI_API_KEY"]
+            # Sync endpoint copy if present
+            try:
+                from tldw_Server_API.app.api.v1.endpoints import chat as chat_endpoint
+                setattr(chat_endpoint, "API_KEYS", chat_schemas.API_KEYS)
```

Ruff (0.14.4)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)